client/modules/User/components/SignupForm.jsx

```diff
@@ -24,10 +24,17 @@ function asyncValidate(fieldToValidate, value) {
       return '';
     });
 }
-
+let timeout=0
 function validateUsername(username) {
-  return asyncValidate('username', username);
-}
+   clearTimeout(timeout)
+ let timeout= setTimeout(function(){
```

Comment:
You don't want a `let` here because that will create a separate `timeout` variable that is scoped to within the body of the `validateUsername` function, rather than re-using the one that you declared outside the function body.